### PR TITLE
Implement new-style mixin support for Dart 2.1

### LINKED ISSUE #1752: support the new super mixins
The overall tracking issue is here: https://github.com/dart-lang/language/issues/12.

This is the dartdoc specific issue for supporting the new super mixins feature.

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #1752, #1778, #1779.

This drops support for Dart 2.0.

As a side effect of this change, type inference is more accurate for deep inheritance cases and class members are more likely to correctly pick up documentation from superclasses.   Additionally, some warnings WRT comment references inherited from superclasses referencing renamed type and parameter variables in subclasses were accidentally suppressed, and are not suppressed anymore (approximately doubling the raw number of warnings in the flutter repository).

Obligatory screenshots:
screenshot from 2018-09-27 15-17-13

and:

screenshot from 2018-09-27 15-15-59

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,6 @@ language: dart
 sudo: false
 dart:
   - "dev/raw/latest"
-  - stable
```
Comment (devoncarew) [inline .travis.yml]: We may not need to omit this; I think at least the analyzer would be happy with both 2.01 and 2.1, but your call here.

**Comment by devoncarew** [comment]
@bwilkerson, can you review this?

**Comment by jcollins-g** [comment]
Reenabled stable tests in travis and changed version range to allow 2.0.  We still require a version override due to https://github.com/dart-lang/build/pull/1873 but we can hack around that to publish.

File: .travis.yml (hunk 1 above)
Comment (jcollins-g) [reply .travis.yml]: Restored testing with stable, as it seems to work.

File: lib/src/model.dart [hunk 2]
```diff
@@ -2681,7 +2776,11 @@ abstract class ModelElement extends Canonicalization
         // Also handles enums
         if (e is ClassElement) {
           if (!e.isEnum) {
-            newModelElement = new Class(e, library, packageGraph);
+            if (e is MixinElementImpl) {
```
[the remark this answers is not in the document]
Comment (jcollins-g) [reply lib/src/model.dart]: Done

File: lib/src/model.dart [hunk 3]
```diff
@@ -2815,8 +2914,14 @@ abstract class ModelElement extends Canonicalization
         .packageGraph.libraryElementReexportedBy[this.element.library];
   }
 
-  // TODO(jcollins-g): annotations should now be able to use the utility
-  // functions in package for finding elements and avoid using computeNode().
+  AstNode _astNode;
+  AstNode get astNode {
+    if (_astNode == null) {
+      _astNode = element?.computeNode();
+    }
```
[the remark this answers is not in the document]
Comment (jcollins-g) [reply lib/src/model.dart]: done

File: lib/src/model.dart [hunk 4]
```diff
@@ -3314,12 +3441,14 @@ abstract class ModelElement extends Canonicalization
   @override
   String get name => element.name;
 
+  // TODO(jcollins-g): refactor once mixins can call super
```
[the remark this answers is not in the document]
Comment (jcollins-g) [reply lib/src/model.dart]: Clarified to make it clear that we have to go to Dart 2.1+ only to refactor this code.  Head dartdoc still can run on Dart 2.0.

**Pull request merged**